scripts/test-check-false-positives.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+cleanup() {
+  r=$?
+
+  # Dump stack logs
+  elastic-package stack dump -v --output "build/elastic-stack-dump/check-${PACKAGE_UNDER_TEST:-${PACKAGE_TEST_TYPE:-*}}"
+
+  # Take down the stack
+  elastic-package stack down -v
+
+  # Clean used resources
+  for d in test/packages/${PACKAGE_TEST_TYPE:-false_positives}/${PACKAGE_UNDER_TEST:-*}/; do
+    (
+      cd $d
+      elastic-package clean -v
+    )
+  done
+
+  # This is a false positive scenario and tests that the test case failure is a success scenario  
+  if [ "${PACKAGE_TEST_TYPE:-false_positives}" == "false_positives" ]; then
+    if [ $r == 1 ]; then
+        exit 0
+      elif [ $r == 0 ]; then
+        exit 1
+    fi
```

Comment:
Currently, with these changes even if the new step runs successfully, there are some JUnit annotations because there were some expected tests failing.

To avoid showing those Junit annotations in Buildkite, I think one option that could we do here is to delete the `build/test-results/*.xml` files in the cleanup function if "$r == 1" . WDYT ? @bhapas 

![Junit annotations with errors](https://github.com/elastic/elastic-package/assets/5330827/e9ffe917-161f-44ce-9002-fd83a62aff1a)

Also, in case there were no errors, some echo message could be added there to explain why it is failing.


```suggestion
    if [ $r == 1 ]; then
      rm build/test-results/*.xml
      exit 0
    elif [ $r == 0 ]; then
      echo "Expected to fail tests, but there was none failing"
      exit 1
    fi
```

cc @jsoriano